tests/test_v_configuration.py

```diff
@@ -0,0 +1,29 @@
+from test_utils import session, normalize_spaces
+from unittest.mock import patch, MagicMock
+from streamlit.testing.v1 import AppTest
+
+def sql_handler(*args):
+    query_result = MagicMock()
+    stmt = normalize_spaces(args[0])
+
+    if stmt == 'select is_first_time_setup_dismissed from config_data.configuration':
+        query_result.collect.return_value = [{'IS_FIRST_TIME_SETUP_DISMISSED': True}]
+    elif stmt == 'select enable_warning_generation_task from config_data.configuration limit 1':
+        query_result.collect.return_value = [{'ENABLE_WARNING_GENERATION_TASK': True}]
+    else:
+        raise NotImplementedError(f'"{stmt}"')
+    
+    return query_result
+
+@patch('snowflake.snowpark.session.Session.builder')
+def test_streamlit_configuration_ui(builder, session):
+    builder.getOrCreate.return_value = session
```

Comment:
anything preventing us from also patching Session.builder in the fixture, and wiring them by default so that getOrCreate returns the one session?

Reply:
My understanding it can't work (I tried it) because of the patch scope. If I put the patch in the fixture, its scope doesn't apply to the test anymore. If you know of a way to do it, please let me know.

Reply:
Something like this?

```
from unittest.mock import patch

@pytest.fixture
def session_builder(session):
    with patch('snowflake.snowpark.session.Session.builder', MagicMock()) as builder:
       builder.getOrCreate.return_value = session
        yield
```

disclaimer: I haven't tried it, just typing here from a guess

Reply:
It works. Thanks